src/evaluators/CallExpression.js

```diff
@@ -78,14 +79,23 @@ function callBothFunctionsAndJoinTheirEffects(args: Array<Value>, ast: BabelNode
 
   // return or throw completion
   if (completion instanceof AbruptCompletion) throw completion;
-  invariant(completion instanceof NormalCompletion || completion instanceof Value || completion instanceof Reference);
+  invariant(completion instanceof NormalCompletion || completion instanceof Value);
   return completion;
 }
 
-function EvaluateCall(ref: Value | Reference, func: Value, ast: BabelNodeCallExpression, strictCode: boolean, env: LexicalEnvironment, realm: Realm): Completion | Value | Reference {
-  if (func instanceof AbstractValue && func.getType() === FunctionValue) {
-    if (func.kind === "conditional")
+function EvaluateCall(
+  ref: Value | Reference, func: Value, ast: BabelNodeCallExpression,
+  strictCode: boolean, env: LexicalEnvironment, realm: Realm
+): Completion | Value | Reference {
+  if (func instanceof AbstractValue) {
+    if (func.getType() !== FunctionValue) {
+      let loc = ast.callee.type === "MemberExpression" ? ast.callee.property.loc : ast.callee.loc;
+      let error = new CompilerDiagnostics("might not be a function", loc, 'PP0005', 'RecoverableError');
+      if (realm.handleError(error) === 'Fail') throw fatalError;
+    } else if (func.kind === "conditional") {
       return callBothFunctionsAndJoinTheirEffects(func.args, ast, strictCode, env, realm);
+    }
+    // Assume that it is a safe function. TODO: really?
```

Comment:
Mention issue number that this relates to: https://github.com/facebook/prepack/issues/705